Mathlib/SetTheory/Cardinal/Finsupp.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Mario Carneiro, Floris van Doorn
```

Comment:
Most of the theorems in this file were proven by [me](https://github.com/leanprover-community/mathlib3/pull/15198/files) and [Junyan Xu](https://github.com/leanprover-community/mathlib3/pull/15889).